File: Backend/Mono/Nonagon.Modular.DynamicForm/Template/Table.cs
using System;
using System.Collections.Generic;

namespace Nonagon.Modular.DynamicForm.Template
{
	/// <summary>
	/// Table layout.
	/// </summary>
	public class Table : FormElementGroup, IHasChildren
	{
		/// <summary>
		/// Gets or sets a value indicating whether this instance has borders.
		/// </summary>
		/// <value><c>true</c> if this instance has borders; otherwise, <c>false</c>.</value>
		public Boolean HasBorders { get; set; }

		/// <summary>
		/// Gets or sets the column count.
		/// </summary>
		/// <value>The column count.</value>
		public Int32 ColumnCount { get; set; }
	}
}

File: Frontend/GWT/gwt-modular-dynamicform/src/com/nonagon/modular/dynamicform/client/template/Table.java
package com.nonagon.modular.dynamicform.client.template;

import com.google.gwt.core.client.JavaScriptObject;

public class Table extends FormElementGroup {
	
	public static Table create() {
		
		Table instance = ((Table)JavaScriptObject.createObject());
		instance.init();
		
		return instance;
	}
	
	protected final native void init()/*-{
		
		this.__type = "Nonagon.Modular.DynamicForm.Template.Table, Nonagon.Modular.DynamicForm";
	}-*/;

	protected Table() {}
	
	public final native boolean getHasBorders()/*-{ return this.HasBorders; }-*/;
	public final native void sethasBorders(boolean value)/*-{ this.HasBorders = value; }-*/;
	
	public final native int getColumnCount()/*-{ return this.ColumnCount; }-*/;
	public final native void setColumnCount(int value)/*-{ this.ColumnCount = value; }-*/;
}
